### PR TITLE
Revert the new "Offer Jetpack Complete" Jetpack post-connection page.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
**TLDR;**
This PR reverts the **PT: Offer Jetpack Complete immediately after the main plugin activation** ( p1HpG7-jYo-p2 ) Jetpack plugin post-connection page/project and sets it back to the Jetpack Cloud Pricing page (the `wordpress.com/jetpack/connect/plans` version).
<hr/>
Upon a thorough analysis done by @simonktnga8c via P2 post: "**The new Jetpack Complete card during signup &mdash; how is it performing?**" ( pbNhbs-5JE-p2 ), it was found that this change/project resulted in poor/limited performance and therefore it was decided to revert back ( pbNhbs-5JE-p2#comment-14031 ) to the previous post-connection page (The Jetpack Cloud Pricing page).

.
### Implementation details:

Fortunately, the new "Offer Jetpack Complete" post-connection page was implemented behind the `jetpack/offer-complete-after-activation` feature-flag in Calypso, allowing us to easily enable and/or disable it.
This PR disables (sets the value to `false`) the `jetpack/offer-complete-after-activation` feature-flag in all Calypso blue (wordpress.com) environments.
These environments are defined in 6 files:
- config/production.json
- config/stage.json
- config/horizon.json
- config/wpcalypso.json
- config/test.json (This one is already set to "false")
- config/development.json

Asana Task: 1198218726984184-as-1204195289136223/f

## Testing Instructions

**Prerequisites:**
Make sure you are logged into your A18N wordpress.com account.

**Test this PR**
- Git checkout this PR/branch and spin up Calypso blue: `git fetch && git checkout revert/offer-jetpack-complete-post-connection-page  && yarn start`
- Create Jurassic Ninja site.
- Connect Jetpack: Click the "Set up Jetpack" button.
- You should be taken to the authorization page. On this page, in the URL, replace `https://wordpress.com` with `http://calypso.localhost:3000` & reload the page.
- Click the "Approve" button.
- Verify you are redirected to the Jetpack connect pricing page: http://calypso.localhost:3000/jetpack/connect/plans/:site?various-query-args&somekey=someval&etc

**To view previous connection behavior before this PR:**
- On the pricing page, click "Start for free", so then you should be redirected back to the Jetpack plugin.
- Now go to the wp-admin Jetpack Dashboard. (Jetpack --> Dashboard). Scroll down to the "Site connection" section and click "Manage site connection" --> "Disconnect" --> "No thank you".
- Click the "Set up Jetpack" button again.
- This time on the authorization page, just click "Approve" _without_ changing the URL to calypso.localhost:3000 (staying in production).
- Verify you are redirected to the new "Offer Jetpack Complete" post-connection page.

## Pre-merge Checklist

- [X ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by manzoorwanijk — APPROVED** [review]
Works as expected

**Review by IanRamosC — APPROVED** [review]
Tests well!

**Comment by andrii-lysenko** [comment]
@elliottprogrammer, just wanted to remind that this will break E2E tests in Jetpack monorepo and we would either need to fix them or give a heads up.

**Comment by elliottprogrammer** [comment]
> @elliottprogrammer, just wanted to remind that this will break E2E tests in Jetpack monorepo and we would either need to fix them or give a heads up.

Ohhh, right!  Thanks for the heads up @andrii-lysenko! Do you know which test it is? or know which PR set the test to assert that the Jetpack Complete page is shown after plugin connection (instead of the Jetpack plans page)?

Oh, I think maybe it's this one: https://github.com/Automattic/jetpack/pull/29284 that @dereksmart implemented. @dereksmart, do you know what's the best way to handle this?  Is it possible to temporarily disable this e2e test so we can ship this PR (which will display the Jetpack plans post-connection page instead of the Jetpack Complete post-connection page.) And then we can fix the e2e test back to the way it was previously?  Is this something you can help with? I'm not quite sure how to handle changes like this (and with dependent plugin e2e tests) because of Jetpack release dates and so forth.  Any assistance here is much appreciated, if possible. Thanks much! 🙂

**Comment by elliottprogrammer** [comment]
Ok, per @dereksmart in https://github.com/Automattic/jetpack/pull/29284#issuecomment-1476292833 , it's totally fine to come back and fix/update the e2e tests after deploying this PR.
E2E tests are non-blocking.
🙂

This PR is good to go! 🚀
And I've created a new Asana task to Fix/revert the Jetpack E2E test here: 1198218726984184-as-1204212388122282/f

**Comment by tyxla** [comment]
It's been 2 years! Time to clean it up: #99375
